Dockerfile

```diff
@@ -1,14 +1,31 @@
 FROM node:12-alpine
 WORKDIR /usr/src/app
 
-COPY . .
+ENV NODE_ENV=development \
+    PORT=3000 \
+    NEO4J_URL= \
+    NEO4J_USERNAME= \
+    NEO4J_PASSWORD= \
+    JWT_AUTH_KEY= \
+    FILES_S3_BUCKET=
 
+# Setup server env dependencies
 RUN apk --no-cache --update --virtual build-dependencies add build-base python
 
-RUN yarn
+# Setup deterministic npm module installation
+# We copy the package.json and yarn.lock separately to ensure Docker build cache is busted only when these files change
+ADD package.json yarn.lock /tmp/
+RUN cd /tmp && yarn
+
+# Setup application code and symlink to node modules
+RUN mkdir -p /opt/cord-api && cd /opt/cord-api && ln -s /tmp/node_modules
+COPY . /opt/cord-api
+WORKDIR /opt/cord-api
+
+RUN yarn build
 
 RUN apk del build-dependencies
 
-EXPOSE 3000
+EXPOSE $PORT
```

Comment:
Why not standard port 80? It can be mapped when running right?

Reply:
We seem to be using port 3000 by default, I didn't want to mess up any of Dmitiry's deployments because he's already mapping to the container's port 3000 in Fargate. Happy to change this to 80 if Dimitry is open to updating the port mappings in Fargate first!

Reply:
@chaurov What do you think? Can you change? Port 80 is the standard web server port.

Reply:
I can change, if needed